### PR TITLE
Combine portable and system builds

### PR DESCRIPTION
**Pull request by Scrumplex**
Portable builds now have the same layout as system builds. If you want to build a portable bundle, you now need to additionally install the `portable` component. In any case the data directory can now be overridden to be portable by creating a `portable.txt` in the root of the application.

For example:

```sh
$ cmake -Bbuild -DCMAKE_INSTALL_PREFIX=install ...
$ cmake --build build
$ cmake --install build
$ cmake --install build --component portable
```

# TODO
- [ ] Do we want to offer different downloads for portable and system builds?
- [ ] Write a better explanation in `program_info/portable.txt`
- [x] Exclude this feature on macOS

### CODE REVIEW

**Comment by ZekeZDev** [comment]
I think different downloads is probably a good idea.

**Comment by Scrumplex** [comment]
> I think different downloads is probably a good idea.

We could also just tell users, that if they want a portable build they can just create a `portable.txt` inside their downloaded archive. That would also avoid confusion once people want to update and don't really know if they were using a portable build before or not.

Edit: I would argue that most people wouldn't want a portable build

**Comment by txtsd** [comment]
I was averse to this way of doing it because the `portable.txt` thing isn't really a standard, but at least one user has asked for it so it works, I suppose.

Different downloads should be okay for now so users know what they're getting without a hassle. We can unify them a few versions later. We should add it to our FAQ either way, once this change is in a release.

File: launcher/Application.cpp [hunk 1]
```diff
@@ -324,30 +344,32 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
...
-        adjustedBy += "Fallback to binary path " + dataPath;
-#endif
+
+#ifndef Q_OS_MACOS
+        if (QFile::exists(FS::PathCombine(m_rootPath, "portable.txt"))) {
+            dataPath = m_rootPath;
+            adjustedBy = "Portable data path";
+        }
     }
+#endif
```
Comment (Scrumplex) [inline launcher/Application.cpp:371]: Oops

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -115,174 +102,175 @@ jobs:
...
+      ##
+
+      - name: Package (macOS)
+        if: runner.os == 'macOS'
         run: |
           cmake --install ${{ env.BUILD_DIR }}
 
-      - name: Install on Windows
+          cd ${{ env.INSTALL_DIR }}
+          macdeployqt "PolyMC.app" -executable="PolyMC.app/Contents/MacOS/polymc" -always-overwrite
```
Comment (DioEgizio) [inline .github/workflows/build.yml]: actually, macdeployqt can be removed completely here

Comment (DioEgizio) [reply .github/workflows/build.yml]: cmake already did everything we needed anyways

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -115,174 +102,175 @@ jobs:
...
+            cp /mingw32/bin/libcrypto-1_1.dll /mingw32/bin/libssl-1_1.dll ./
+          elif [ "${{ matrix.msystem }}" == "mingw64" ]; then
+            cp /mingw64/bin/libcrypto-1_1-x64.dll /mingw64/bin/libssl-1_1-x64.dll ./
+          fi
+
+      - name: Package (Windows, portable)
+        if: runner.os == 'Windows'
+        shell: msys2 {0}
+        run: |
+          cmake --install ${{ env.BUILD_DIR }} --prefix ${{ env.INSTALL_PORTABLE_DIR }}
```
Comment (DioEgizio) [inline .github/workflows/build.yml]: i think here you should copy what you've done before but adding the portable stuff.
The current way installs two times, that makes the windows action even slower

Comment (Scrumplex) [reply .github/workflows/build.yml]: Sounds like a good idea. Forgot about the slow install step on Windows

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -115,174 +102,175 @@ jobs:
...
+          cd ${{ env.INSTALL_DIR }}
+          tar -czf ../PolyMC.tar.gz *
 
-      - name: Install on Linux portable
-        if: runner.os == 'Linux' && matrix.portable == true
+      - name: Package (Linux, portable)
+        if: runner.os == 'Linux'
         run: |
-          cmake --install ${{ env.BUILD_DIR }}
+          cmake --install ${{ env.BUILD_DIR }} --prefix ${{ env.INSTALL_PORTABLE_DIR }} --component portable
```
Comment (DioEgizio) [inline .github/workflows/build.yml:194]: same thing here, but you've done an oopsie and it just contains the portable.txt

Comment (DioEgizio) [reply .github/workflows/build.yml:194]: immagine

Comment (Scrumplex) [reply .github/workflows/build.yml:194]: lol

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -115,174 +102,170 @@ jobs:
...
-      - name: Install on Linux portable
-        if: runner.os == 'Linux' && matrix.portable == true
+      - name: Package (Linux, portable)
+        if: runner.os == 'Linux'
         run: |
-          cmake --install ${{ env.BUILD_DIR }}
+          cmake --install ${{ env.BUILD_DIR }} --prefix ${{ env.INSTALL_PORTABLE_DIR }}
+          cmake --install ${{ env.BUILD_DIR }} --prefix ${{ env.INSTALL_PORTABLE_DIR }} --component portable
+
+          cd ${{ env.INSTALL_PORTABLE_DIR }}
```
Comment (DioEgizio) [inline .github/workflows/build.yml:196]: whitespace

Comment (Scrumplex) [reply .github/workflows/build.yml:196]: actually intended

Comment (DioEgizio) [reply .github/workflows/build.yml:196]: oh

**Review by txtsd — APPROVED** [review]

**Comment by ZekeZDev** [comment]
I still dont like this, I think seperate downloads on the website make more sense. Are we able to still have seperate downloads with this pr?

**Comment by DioEgizio** [comment]
> I still dont like this, I think seperate downloads on the website make more sense. Are we able to still have seperate downloads with this pr?

If you looked better, it has separate downloads

**Review by ZekeZDev — APPROVED** [review]

**Review by d-513 — APPROVED** [review]

**Comment by DioEgizio** [comment]
this is conflicting with develop

**Comment by txtsd** [comment]
Needs `-DENABLE_LTO=ON` from #333

**Comment by Scrumplex** [comment]
Please re-review.

**Review by txtsd — APPROVED** [review]
LGTM!

**Review by DioEgizio — APPROVED** [review]
lgtm

**Pull request merged**